src/project_root.rs

```diff
@@ -24,3 +35,31 @@ pub fn find_project_root(roots: &[String], path: &str) -> String {
         }
     }
 }
+
+pub fn gather_env_roots(language: &str) -> HashSet<PathBuf> {
+    let prefix = format!("KAK_LSP_PROJECT_ROOT_{}", language.to_uppercase());
+    debug!("Searching for vars starting with {}", prefix);
+    let p = prefix.as_str();
```

Comment:
nit: I didn't check, but it should be enough just to use `&prefix` below without explicit conversion.